src/apps/material/helper.ts

```diff
@@ -146,13 +147,101 @@ export const totalBranchesHaveMaterial = (
   }).length;
 };
 
-export const getInfomediaId = (manifestation: Manifestation) => {
-  const access = manifestation?.access || [];
-  return access.reduce<string | null>((acc, curr) => {
-    const { __typename: type } = curr;
-    if (type === "InfomediaService") {
-      return curr.id;
+export const getInfomediaIds = (manifestations: Manifestation[]) => {
+  const infomediaIds = manifestations
+    .map((manifestation) =>
+      manifestation.access.map((currentAccess) => {
+        return currentAccess.__typename === "InfomediaService"
+          ? currentAccess.id
+          : null;
+      })
+    )
+    .flat();
+  return compact(infomediaIds);
+};
+
+export const getAllUniqueMaterialTypes = (manifestations: Manifestation[]) => {
+  const allMaterialTypes = manifestations
+    .map((manifest) => manifest.materialTypes.map((type) => type.specific))
+    .flat();
+  return uniq(allMaterialTypes);
+};
+
+export const divideManifestationsByMaterialType = (
+  manifestations: Manifestation[]
+) => {
+  const uniqueMaterialTypes = getAllUniqueMaterialTypes(manifestations);
+  const dividedManifestationsArrays = uniqueMaterialTypes.map(
+    (uniqueMaterialType) => {
+      return manifestations.filter((manifest) => {
+        const manifestationMaterialTypes = manifest.materialTypes.map(
+          (materialType) => materialType.specific
+        );
+        return manifestationMaterialTypes.includes(uniqueMaterialType);
+      });
     }
-    return acc;
-  }, null);
+  );
+  const dividedManifestationsObject: { [key: string]: Manifestation[] } =
+    dividedManifestationsArrays.reduce((result, current, index) => {
+      const materialType = uniqueMaterialTypes[index];
+      return { ...result, [materialType]: current };
+    }, {});
+  return dividedManifestationsObject;
+};
+
+export const getAllIdentifiers = (manifestations: Manifestation[]) => {
+  return manifestations
+    .map((manifestation) =>
+      manifestation.identifiers.map((identifier) => identifier.value)
+    )
+    .flat();
+};
+
+export const getAllPids = (manifestations: Manifestation[]) => {
+  return manifestations.map((manifestation) => manifestation.pid);
+};
+
+export const getManifestationsWithMaterialType = (
+  manifestations: Manifestation[]
+) => {
+  return manifestations.filter((manifestation) => {
+    return manifestation.materialTypes.length > 0;
+  });
+};
+
+export const isABook = (manifestations: Manifestation[]) => {
+  return manifestations.some((manifestation) => {
+    return manifestation.materialTypes.some(
+      (materialType) =>
+        materialType.specific.toLowerCase() === MaterialType.book
+    );
+  });
+};
+
+export const getBestMaterialTypeForManifestation = (
+  manifestation: Manifestation
+) => {
+  if (isABook([manifestation])) {
```

Comment:
I think it would be nice with a comment stating the reason why a book has precedence over other types

Reply:
That's actually a great question - why is it? I just always hear Tue talking about how they're lucky book comes first in the alphabet.